source/docs/software/networktables/creating-sendable-classes.rst

```diff
@@ -0,0 +1,163 @@
+
+To expose information about a class to users over NetworkTables, one must create an implementation of the initSendable() method of the Sendable interface. The properties added to Sendable classes require a getter and setter for the property. Note that a :code:`null` setter simply means the value can't be changed over NetworkTables, and may be desired in situations where information should only be displayed, but not modified. If the Sendable is configured as an actuator, the setter will be called when the robot is enabled in Test mode when users change its values to test robot functions. Many of WPI's classes, such as :code:`Command` and :code:`Subsystem`, already implement Sendable. To add new data to a Sendable class, such as adding a motor or sensor specific to a Subsystem, teams can simply override the supertype's :code:`initSendable` method. Note that teams must still call the :code:`super.initSendable` method to run the superclass' implementation. For example, Command's implementation of :code:`Sendable` includes code that will configure SmartDashboard or Shuffleboard to display it as a command, and allows users to start and stop the command through these dashboards.
+
+.. tabs::
+
+      .. code-tab:: c++
+
+        void MySubsystem::InitSendable(SendableBuilder& builder) {
+
+            builder.AddDoubleProperty(".intakeSpeed",
+                    [=]() { return
+                        myMotor.Get(); },
+                    [=](double value) {
+                        myMotor.Set(value);
+                });
+
+                Subsystem.InitSendable(builder);
+
+        }
+
+    .. code-tab:: java
+
+        public class MySubsystem extends Subsystem {
+
+            Spark spark = new Spark(1);
+            Solenoid solenoid = new Solenoid(0);
+
+            @Override
+            public void initSendable(SendableBuilder builder) {
+
+               builder.addDoubleProperty(".intakeSpeed", spark::get, (value) -> spark.set(value));
+
+               // this call sets up Command specific configuration, including calling setSmartDashboardType("Subsystem")
+               // to let Shuffleboard know to treat this as such.
+               super.initSendable(builder);
+            }
+
+        }
+
+Creating a new Sendable class with SendableBuilder
+--------------------------------------------------
+
+Creating an entirely new :code:`Sendable` class follows a similar pattern to the extending the implementation present in subclasses of WPI classes such as :code:`Command` and :code:`Subsystem` as presented above. The main difference is that when creating a new class, teams must explicitly define the Sendable's type. This allows teams to create custom wigets for Shuffleboard to display data. This example shows a data class which holds boolean and double values about an arm's state.
+
+.. tabs::
+
+    .. code-tab:: c++
+
+        TOOD
+
+    .. code-tab:: java
+
+        public class ArmState implement Sendable {
```

Comment:
An immutable data class is a bad example for a `Sendable` implementation, since the properties are only set once - users would have to send each new instance to the dashboard after creating it, which is not very efficient (and may not even be allowed by the dashboard framework).

If you want to continue with this route, I suggest changing the constructor to take an `ArmSubsystem` parameter and have `initSendable` call the appropriate methods on the subsystem (eg `arm::getAngle`, `arm::isOpen`, etc)